src/main/webapp/assets/javascript/app.js

```diff
@@ -0,0 +1,252 @@
+            const json = response.json();
+            //   console.log(json);
+            return json;
+        }
+        todoListData = request();
+    }
+
+    getToDo();
+
+    // Check genreIDs and genre names:
+    // http://api.themoviedb.org/3/movie/:movieID?api_key=<<>>
+    //28 = action
+    //12 = adventure
+    //16 = animation
+    //35 = comedy
+    //80 = crime
+    //18 = drama
+    //10751 = family
+    //14 = fantasy
+    //36 = history
+    //27 = horror
+    //10402 = music
+    //10749 = romance
+    //878 = science fiction
+    //53 = thriller
+
+    function inToDoList(EntityType, EntityID) {
+        console.log(EntityType + " " + EntityID);
+        for (let i = 0; i < todoListData.length; i++) {
+            //   console.log(EntityID + " " + todoListData[i].id);
+            if (todoListData[i].id == EntityID && todoListData[i].type == EntityType) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    function getModalCode(obj, i, entityType) {
+        var id = obj.results[i].id;
+        var poster = imageBaseUrl + 'w300' + obj.results[i].poster_path;
+        var title = obj.results[i].original_title;
+        // var youtubeKey = movieKey.results[0].key;
+        // var youtubeLink = 'https://www.youtube.com/watch?v=' + youtubeKey;
+        var releaseDate = obj.results[i].release_date;
+        var overview = obj.results[i].overview;
+        var voteAverage = obj.results[i].vote_average;
+        var codeHTML = '';
+        codeHTML += '<div class="col-sm-3 eachMovie">';
+        codeHTML += '<button type="button" class="btnModal" data-toggle="modal" data-target="#exampleModal' + obj.results[i].id + '" data-whatever="@' + obj.results[i].id + '">' + '<img src="' + poster + '"></button>';
+        codeHTML += '<div class="modal fade" id="exampleModal' + obj.results[i].id + '" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">';
+        codeHTML += '<div class="modal-dialog" role="document">';
+        codeHTML += '<div class="modal-content col-sm-12">';
+        codeHTML += '<div class="col-sm-6 moviePosterInModal">';
+        // codeHTML += '<a href="' + youtubeLink + '"><img src="' + poster + '"></a>';
+        codeHTML += '<img src="' + poster + '">';
+        codeHTML += '</div><br>'; //close trailerLink
+        codeHTML += '<div class="col-sm-6 movieDetails">';
+        codeHTML += '<div class="movieName">' + title + '</div><br>';
+        // codeHTML += '<div class="linkToTrailer"><a href="' + youtubeLink + '"><span class="glyphicon glyphicon-play"></span>&nbspPlay trailer</a>' + '</div><br>';
+        codeHTML += '<div class="release">Release Date: ' + releaseDate + '</div><br>';
+        codeHTML += '<div class="overview">' + overview + '</div><br>'; // Put overview in a separate div to make it easier to style
+        codeHTML += '<div class="rating">Rating: ' + voteAverage + '/10</div><br>';

```

Comment:
can this be a single string.. I am afraid so many concats will bring in in-efficiency